source/base/data_out_base.cc

```diff
@@ -8191,68 +8192,124 @@ DataOutInterface<dim, spacedim>::create_xdmf_entry(
   (void)cur_time;
   (void)comm;
   AssertThrow(false, ExcMessage("XDMF support requires HDF5 to be turned on."));
-#endif
-  AssertThrow(spacedim == 2 || spacedim == 3,
-              ExcMessage("XDMF only supports 2 or 3 space dimensions."));
+
+  return {};
+
+#else
+
+  std::uint64_t local_node_cell_count[2], global_node_cell_count[2];
 
   local_node_cell_count[0] = data_filter.n_nodes();
-  local_node_cell_count[1] = data_filter.n_cells();
+  // n_cells returns an invalid unsigned int if the object is empty:
+  local_node_cell_count[1] =
+    (data_filter.n_nodes() > 0) ? data_filter.n_cells() : 0;
```

Comment:
It's a weird inconsistency that `n_nodes()` returns the 'correct' value while `n_cells()` can return a value initialized to `invalid_unsigned_int`. Should we change the logic behind `n_cells()`?

Reply:
I agree. I will make that change.